### ISSUE TITLE
How to build an alpine-python:3.5 container with ansible-container

### ISSUE THREAD

**Issue by cognitiaclaeves** [ansible/ansible-container#511]
I don't think there are any issues reported for Alpine-python:3.5 yet. Since Ansible needs Python 2.7 right now, building a reduced OS container based solely on python 3.5 appears to present an interesting issue.

##### ISSUE TYPE

 - Bug Report

##### container.yml 

```
settings:
  conductor_base: python:2.7-alpine-plus
services:
  docker_ds_api_monitor:
    roles:
    - docker_ds_api_monitor
```

Dockerfile to create custom conductor container, python:2.7-alpine-plus
```
from python:2.7-alpine
RUN apk --no-cache --update add tar gcc libffi-dev musl-dev openssl-dev sshpass

CMD ["/bin/sh"]
```

```
$ cat roles/docker_ds_api_monitor/meta/container.yml
from: python:3.5-alpine
maintainer: J. Norment
environment:
  PYTHONUNBUFFERED: '0'
user: '{{ user }}'
working_dir: '"{{ script_home }}/run"'
command:
- sh
- -c
- $SCRIPT_HOME/etc/dumb-init arbitrary-python-script.py
```

##### OS / ENVIRONMENT

```
$ sudo ansible-container --debug version                                                                                                          [120/1859]
Ansible Container, version 0.9.0.0
Linux, jn-smart-etl-test, 3.10.0-514.2.2.el7.x86_64, #1 SMP Tue Dec 6 23:06:41 UTC 2016, x86_64
2.7.5 (default, Nov  6 2016, 00:28:07)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-11)] /usr/bin/python2
{
  "ContainersPaused": 0,
  "Labels": null,
  "CgroupDriver": "cgroupfs",
  "ContainersRunning": 1,
  "ContainerdCommit": {
    "Expected": "4ab9917febca54791c5f071a9d1f404867857fcc",
    "ID": "4ab9917febca54791c5f071a9d1f404867857fcc"
  },
  "InitBinary": "docker-init",
  "NGoroutines": 27,
  "Swarm": {
    "Managers": 0,
    "ControlAvailable": false,
    "NodeID": "",
    "Cluster": {
      "Spec": {
        "TaskDefaults": {},
        "Orchestration": {},
        "EncryptionConfig": {
          "AutoLockManagers": false
        },
        "Raft": {
          "HeartbeatTick": 0,
          "ElectionTick": 0
        },
        "CAConfig": {},
        "Dispatcher": {}
      },
      "Version": {},
      "ID": "",
      "CreatedAt": "0001-01-01T00:00:00Z",
      "UpdatedAt": "0001-01-01T00:00:00Z"
    },
    "Nodes": 0,
    "Error": "",
    "RemoteManagers": null,
    "LocalNodeState": "inactive",
    "NodeAddr": ""
  },
  "LoggingDriver": "json-file",
  "OSType": "linux",
  "HttpProxy": "",
  "Runtimes": {
    "runc": {
      "path": "docker-runc"
    }
  },
  "DriverStatus": [
    [
      "Backing Filesystem",
      "xfs"
    ],
    [
      "Supports d_type",
      "true"
    ]
  ],
  "OperatingSystem": "CentOS Linux 7 (Core)",
  "Containers": 113,
  "HttpsProxy": "",
  "BridgeNfIp6tables": false,
  "MemTotal": 1757741056,
  "SecurityOptions": [
    "name=seccomp,profile=default"
  ],
  "Driver": "overlay",
  "IndexServerAddress": "https://index.docker.io/v1/",
  "ClusterStore": "",
  "InitCommit": {
    "Expected": "949e6fa",
    "ID": "949e6fa"
  },
"Isolation": "",                                                                                                                                                                               [42/1859]
  "SystemStatus": null,
  "OomKillDisable": true,
  "ClusterAdvertise": "",
  "SystemTime": "2017-05-09T20:09:57.839540327Z",
  "Name": "jn-smart-etl-test",
  "CPUSet": true,
  "RegistryConfig": {
    "InsecureRegistryCIDRs": [
      "127.0.0.0/8"
    ],
    "IndexConfigs": {
      "virt-ds-docker.aus-bin-prd-00.q2dc.local": {
        "Official": false,
        "Name": "virt-ds-docker.aus-bin-prd-00.q2dc.local",
        "Secure": false,
        "Mirrors": []
      },
      "ds-docker.aus-bin-prd-00.q2dc.local": {
        "Official": false,
        "Name": "ds-docker.aus-bin-prd-00.q2dc.local",
        "Secure": false,
        "Mirrors": []
      },
      "docker.io": {
        "Official": true,
        "Name": "docker.io",
        "Secure": true,
        "Mirrors": null
      }
    },
    "Mirrors": []
  },
  "DefaultRuntime": "runc",
  "ContainersStopped": 112,
  "NCPU": 1,
  "NFd": 22,
  "Architecture": "x86_64",
  "KernelMemory": true,
  "CpuCfsQuota": true,
  "Debug": false,
  "ID": "FPGA:53XQ:4SPQ:PFE5:TQLL:VJHU:DMV7:ZUOX:WWHT:QHQ2:QFTN:DPS6",
  "IPv4Forwarding": true,
  "KernelVersion": "3.10.0-514.2.2.el7.x86_64",
  "BridgeNfIptables": false,
  "NoProxy": "",
  "LiveRestoreEnabled": false,
  "ServerVersion": "17.03.1-ce",
  "CpuCfsPeriod": true,
  "ExperimentalBuild": false,
  "MemoryLimit": true,
  "SwapLimit": true,
  "Plugins": {
    "Volume": [
      "local"
    ],
    "Network": [
      "bridge",
      "host",
      "macvlan",
      "null",
      "overlay"
    ],
    "Authorization": null
  },
  "Images": 1156,
  "DockerRootDir": "/var/lib/docker",
  "NEventsListener": 0,
  "CPUShares": true,
  "RuncCommit": {
    "Expected": "54296cf40ad8143b62dbcaa1d90e520a2136ddfe",
    "ID": "54296cf40ad8143b62dbcaa1d90e520a2136ddfe"
  }
}
{
  "KernelVersion": "3.10.0-514.2.2.el7.x86_64",
  "Arch": "amd64",
  "BuildTime": "2017-03-27T17:05:44.088559571+00:00",
  "ApiVersion": "1.27",
  "Version": "17.03.1-ce",
  "MinAPIVersion": "1.12",
  "GitCommit": "c6d412e",
  "Os": "linux",
  "GoVersion": "go1.7.5"
}
```
##### SUMMARY
Unable to build an alpine-based python 3.5 container.
( Ansible needs py 2.7, so conductor needs to be 2.7, but container actually needs to be 3.5, so conductor binary doesn't/can't easily match the container ... )

##### STEPS TO REPRODUCE

```
ansible-container build
```

##### ACTUAL RESULTS

```

```

Really, it would be enough just to see how ansible-container can build any python:3.5-alpine.

**Comment by chouseknecht** [comment]
@cognitiaclaeves, thanks for using Ansible Container, and for taking the time to open an issue to raise this topic. We appreciate your time and effort. 

I'm not completely sure what you're asking for on this ticket. Let me explain... 

Ansible Container supports Alpine as a conductor (or build) container with python2.7 installed. See the [Dockerfile used to create the build container image](https://github.com/ansible/ansible-container/blob/develop/container/docker/templates/conductor-dockerfile.j2).

By default it will share the `/usr` directory from the conductor to the target service container, which obviously prevents the installation of Python 3.5 into the target service container.

If you require Python 3.5 in that container, then use the `--use-local-python` option during the `build`. This will stop it from sharing `/usr`. Then use a `raw` action at the start of the first role to install the Python run-time. You will also want to use the dictionary, or mapping, role specification style, and set `gather_facts: no` for that role.

Also note, the `--use-local-python` option is only available via running Ansible Container from source. See the [running from source topic](http://docs.ansible.com/ansible-container/installation.html#running-from-source) at our docs site.

Hopefully this helps make things clearer.

--Chris

**Comment by cognitiaclaeves** [comment]
Thanks for the response, Chris.

I think I can understand how you might be having difficulties understanding
what I'm asking for, as I'm having difficulties interpreting the answers
I'm getting -- so it's pretty clear that we're on different wavelengths,
somehow. ;-)

Though if I provided code, and you're still not seeing where I'm coming
from, I'm guessing I didn't include enough for you to run it?

Here's where I think my use-case is diverging from the answers I'm seeing
that relate to this topic:

The container for python:3.5-alpine should already have a python runtime
included. So that means that I don't need to build it. But every time this
topic is mentioned, it mentions how to install the runtime.

But the thing that I think is really causing the issue here is
--use-local-python is only available when running Ansible Container from
source. I tried --use-local-python on a pip installation of
ansible-container, and was not successful. That might be why.


On Wed, May 10, 2017 at 8:48 AM, Chris Houseknecht <notifications@github.com
